### DISCUSSION TITLE
PrismaClientInitializationError: Unknown PRISMA_QUERY_ENGINE_LIBRARY rhel-openssl-1.1.x？

### DISCUSSION THREAD

**Discussion by lddlww** [prisma/prisma#18771] [Q&A] [answered]
what's wrong with it？

```
npx prisma -v
Environment variables loaded from .env
prisma                  : 4.12.0
@prisma/client          : 4.12.0
Current platform        : rhel-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at bin/libquery_engine.so.node, resolved by PRISMA_QUERY_ENGINE_LIBRARY)
Migration Engine        : migration-engine-cli 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7 (at bin/migration-engine, resolved by PRISMA_MIGRATION_ENGINE_BINARY)
Format Wasm             : @prisma/prisma-fmt-wasm 4.12.0-67.659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Default Engines Hash    : 659ef412370fa3b41cd7bf6e94587c1dfb7f67e7
Studio                  : 0.483.0
```

**Comment by nurul3101** [comment]
Hello @lddlww 👋 

Have you specified any [binaryTargets](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#binarytargets-options) in your schema file?
On which OS are you generating the PrismaClient?
Can you check if OpenSSL is installed on your machine?

**Reply by lddlww** [reply]
> Have you specified any [binaryTargets](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#binarytargets-options) in your schema file?

no



> On which OS are you generating the PrismaClient?

CentOS Linux release 8.0.1905 (Core)



> Can you check if OpenSSL is installed on your machine?

OpenSSL 1.1.1k  FIPS 25 Mar 2021

**Reply by nurul3101** [reply]
Can you try specifying `binaryTarget` values as `rhel-openssl-1.0.x` as specified [here](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#linux-centos) and check?

**Reply by lddlww** [reply, marked as answer]
i switch to ```Ubuntu 22.04.1 LTS```, it work now

**Reply by nurul3101** [reply]
Thanks for getting back and letting us know!
